tests/foreman/api/test_host.py

```diff
@@ -299,88 +349,114 @@ def test_positive_create_with_inherited_params(self):
         proxy = entities.SmartProxy().search(
             query={'search': 'url = https://{0}:9090'.format(settings.server.hostname)}
         )[0]
         host = entities.Host(puppet_ca_proxy=proxy).create()
-        self.assertEqual(host.puppet_ca_proxy.read().name, proxy.name)
+        assert host.puppet_ca_proxy.read().name == proxy.name
+        new_host = entities.Host().create()
+        new_host.puppet_ca_proxy = proxy
+        new_host = new_host.update(['puppet_ca_proxy'])
+        assert new_host.puppet_ca_proxy.read().name == proxy.name
 
     @tier2
-    def test_positive_create_with_puppet_class(self):
-        """Create a host with associated puppet classes
+    def test_positive_end_to_end_with_puppet_class(
+        self, module_org, module_location, module_env_search, module_puppet_classes
+    ):
+        """Create a host with associated puppet classes then remove it and update the host
+        with same associated puppet classes
 
         :id: 2690d6b0-441b-44c5-b7d2-4093616e037e
 
-        :expectedresults: A host is created with expected puppet classes
+        :expectedresults: A host is created with expected puppet classes then puppet classes
+        are removed and the host is updated with same puppet classes
         """
         host = entities.Host(
-            organization=self.org,
-            location=self.loc,
-            environment=self.env,
-            puppetclass=self.puppet_classes,
+            organization=module_org,
+            location=module_location,
+            environment=module_env_search,
+            puppetclass=module_puppet_classes,
         ).create()
-        self.assertEqual(
-            {puppet_class.id for puppet_class in host.puppetclass},
-            {puppet_class.id for puppet_class in self.puppet_classes},
-        )
+        assert {puppet_class.id for puppet_class in host.puppetclass} == {
+            puppet_class.id for puppet_class in module_puppet_classes
+        }
+        host.puppetclass = []
+        host.environment = []
+        host = host.update(['environment', 'puppetclass'])
+        assert len(host.puppetclass) == 0
+        host.environment = module_env_search
+        host.puppetclass = module_puppet_classes
+        host = host.update(['environment', 'puppetclass'])
+        assert {puppet_class.id for puppet_class in host.puppetclass} == {
+            puppet_class.id for puppet_class in module_puppet_classes
+        }
 
     @tier2
-    def test_positive_create_with_subnet(self):
-        """Create a host with subnet specified
+    def test_positive_create_and_update_with_subnet(self):
+        """Create and update a host with subnet specified
 
         :id: 9aa97aff-8439-4027-89ee-01c643fbf7d1
 
-        :expectedresults: A host is created with expected subnet assigned
+        :expectedresults: A host is created and updated with expected subnet assigned
 
         :CaseLevel: Integration
         """
         org = entities.Organization().create()
```

Comment:
Use global module fixtures instead of creating it from scratch here ?

Reply:
Same for all below but first entities.

Reply:
DONE.